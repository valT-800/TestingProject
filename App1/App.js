import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import OrderScreen from './src/screens/OrderScreen';
import WaitingOrdersScreen from './src/screens/WaitingOrdersScreen';
import HomeScreen from './src/screens/HomeScreen';
import OrdersScreen from './src/screens/OrdersScreen';
import Styles from './Styles'
import WaitingOrderScreen from './src/screens/WaitingOrderScreen';

const Stack = createNativeStackNavigator();

function Navigation(){
  return (
      <NavigationContainer>
        <Stack.Navigator initialRouteName={HomeScreen}>
        <Stack.Screen name='Home' component={HomeScreen} options = {{headerStyle: Styles.header, title: 'Home'}}/>
          <Stack.Screen name='Orders' component={OrdersScreen} options = {{headerStyle: Styles.header, title: 'All orders'}}/>
          <Stack.Screen name = "Waiting orders" component={WaitingOrdersScreen} options = {{headerStyle: Styles.header, title: 'Waiting orders'}}/>
          <Stack.Screen name = "Waiting order" component={WaitingOrderScreen} options = {{headerStyle: Styles.header, title: 'Order details'}}/>   
          <Stack.Screen name = "Order" component={OrderScreen} options = {{headerStyle: Styles.header, title: 'Order details'}}/>   
        </Stack.Navigator>
    </NavigationContainer>
    
  );
}

function App(){
  return(
    <Navigation/>
    
  )
}

export default App;