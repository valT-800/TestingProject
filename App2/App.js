import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import Styles from './Styles';
import ScanScreen from './src/screens/ScanScreen';
import StartScreen from './src/screens/StartScreen';


const Stack = createNativeStackNavigator();

function Navigation(){
  return (
      <NavigationContainer>
        <Stack.Navigator initialRouteName={StartScreen}>
          <Stack.Screen name = "Start" component={StartScreen} options = {{headerStyle: Styles.header, title: ''}}/>
          <Stack.Screen name = 'Scan' component={ScanScreen} options = {{headerStyle: Styles.header, title: ''}}/>
        </Stack.Navigator>
    </NavigationContainer>
    
  );
}


function App(){
  return(
    <Navigation/>
    
  )
}

export default App;