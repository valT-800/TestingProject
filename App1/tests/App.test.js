import { render, screen } from '@testing-library/react-native';
import React from 'react';
import App from '../App';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('Testing react navigation', () => {
  test('Renders snapshot as expected', async () => {

    const tree = render(<App />).toJSON();
    expect(tree).toMatchSnapshot();

  });
  
  });

