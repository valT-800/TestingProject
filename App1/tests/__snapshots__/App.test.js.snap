// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react navigation Renders snapshot as expected 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="turquoise"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "aliceblue",
                "flex": 1,
                "justifyContent": "center",
                "paddingBottom": 20,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "turquoise",
                  "borderColor": "whitesmoke",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingHorizontal": 22,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "darkslategrey",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "textAlign": "center",
                  }
                }
              >
                All orders
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "turquoise",
                  "borderColor": "whitesmoke",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingHorizontal": 22,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "darkslategrey",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "textAlign": "center",
                  }
                }
              >
                Waiting orders
              </Text>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
